geth/jail/jail.go

```diff
@@ -299,5 +300,14 @@ func newJailErrorResponse(err error) string {
 // Marshaling is not required as result.String() produces a string
 // that is a valid JavaScript code.
 func newJailResultResponse(result otto.Value) string {
```

Comment:
Is there any reason not to use `json.Marshal` here?

Reply:
nope, i just got influenced by the function comment, i think it could look something like this
```go
// newJailResultResponse returns a string that is a valid JavaScript code.
func newJailResultResponse(input otto.Value) string {
	type res struct {
		Result interface{} `json:"result"`
	}

	originalValue := input.String()
	val := res{Result: originalValue}
	if newValue, err := strconv.ParseBool(originalValue); err == nil {
		val.Result = newValue
	} else if newValue, err := strconv.ParseFloat(originalValue, 64); err == nil {
		val.Result = newValue
	} else if newValue, err := strconv.ParseInt(originalValue, 10, 64); err == nil {
		val.Result = newValue
	} else if newValue, err := strconv.ParseUint(originalValue, 10, 64); err == nil {
		val.Result = newValue
	}

	result, err := json.Marshal(val)
	if err != nil {
		return newJailErrorResponse(errors.New("Can't marshal result"))
	}
	return string(result)
}
```

Reply:
Definitely that makes more sense, i've changed it :)